### PR TITLE
Feat: Test Lint

### PR DESCRIPTION
**Pull request by CodeWithEmad**
Assorted testing and linting improvements:

* Makefile with test and lint targets
* Long lines reformated with black
* Sort imports with isort

In addition:
* Typing added for various functions

PS. I must admit that the "Typing"  part could use some improvement.

### CODE REVIEW

**Comment by CodeWithEmad** [comment]
@regisb this failed becaue MyPy needs to have `types-requests` installed here. Since we decided that all dev dependencies should be installed from `requirements/dev.txt` in the tutor, is it okay to open a PR there for  `types-requests`?

**Comment by regisb** [comment]
Let's not add types-requests to tutor core if it's not strictly necessary. Can you implement the `[dev]` requirements here, and add types-requests to them, such that we can write in the CI:

    pip install -e .[dev]

instead of:

    pip install 'tutor[dev]>=16.0.0,<17.0.0'

I think this should solve our issue, right?

File: tutorxqueue/plugin.py [hunk 1]
```diff
@@ -2,30 +2,24 @@
 
 import json
 import os
-import typing as t
 from glob import glob
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
```
Comment (regisb) [inline tutorxqueue/plugin.py]: :thinking: not a big fan of that one. Why can't we use `list`, `dict` and `tuple` anymore?

File: tutorxqueue/plugin.py [hunk 2]
```diff
@@ -233,19 +259,38 @@ def grade_submission(self, submission_id, submission_key, grade, correct, msg):
...
+            return None
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+            return None
+        else:
+            try:
+                return response.json()
+            except ValueError:
+                print("Could not parse json")
+            return None
```
Comment (regisb) [inline tutorxqueue/plugin.py]: While I understand that you are trying to handle errors properly, I'd rather keep this in a separate PR for now. That's because error handling can be a tricky affair...

File: tutorxqueue/plugin.py [hunk 3]
```diff
@@ -216,10 +235,17 @@ def show_submission(self, queue):
             "return_code": response["return_code"],
         }
 
-    def count_submissions(self, queue):
+    def count_submissions(self, queue: str) -> Any:
```
Comment (regisb) [inline tutorxqueue/plugin.py:238]: Thanks for taking the time to type all those functions, I understand it's not an easy job, nor a very fun one.

File: tutorxqueue/plugin.py (hunk 1 above)
Comment (CodeWithEmad) [reply tutorxqueue/plugin.py]: There's no reason for that. we can go with old trusty `list`, `dict`, and `tuple`.

File: tutorxqueue/plugin.py (hunk 2 above)
Comment (CodeWithEmad) [reply tutorxqueue/plugin.py]: Agree. I'll take care of that.

File: tutorxqueue/plugin.py (hunk 3 above)
Comment (CodeWithEmad) [reply tutorxqueue/plugin.py:238]: Anytime!

File: setup.py [hunk 4]
```diff
@@ -33,7 +34,13 @@
     packages=find_packages(exclude=["tests*"]),
     include_package_data=True,
     python_requires=">=3.8",
-    install_requires=["tutor>=16.0.0,<17.0.0", "requests"],
+    install_requires=[f"tutor{TUTOR_VERSION}" , "requests"],
+    extras_require={
+        "dev": [
+            f"tutor[dev]{TUTOR_VERSION}" ,
```
Comment (CodeWithEmad) [inline setup.py]: I'm not a fan of this but I didn't want to repeat myself, so I moved the TUTOR_VERSION. let's decide a general form for this, so I can set dev for all plugins.

Comment (CodeWithEmad) [reply setup.py]: after this was merged, should I add a development section for all plugins (and cookiecutter) ?

File: setup.py [hunk 5]
```diff
@@ -33,7 +34,13 @@
...
+            "types-requests==2.31.0.0"
```
Comment (regisb) [inline setup.py]: Let's not pin types-request here.

File: setup.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 from setuptools import setup, find_packages
 
 here = os.path.abspath(os.path.dirname(__file__))
+TUTOR_VERSION = ">=16.0.0,<17.0.0"
```
Comment (regisb) [inline setup.py]: This is too complicated. Let's keep the tutor version inline.

File: setup.py (hunk 5 above)
Comment (CodeWithEmad) [reply setup.py]: installing latest version causes some dependency conflicts.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
google-auth 2.22.0 requires urllib3<2.0, but you have urllib3 2.1.0 which is incompatible.

```

File: setup.py (hunk 6 above)
Comment (CodeWithEmad) [reply setup.py]: Sure.

File: setup.py (hunk 5 above)
Comment (jfavellar90) [reply setup.py]: @CodeWithEmad the error is thrown in [this line](https://github.com/CodeWithEmad/tutor-xqueue/blob/feat/test-lint/tutorxqueue/plugin.py#L10) when not installing types-requests. Could we just ignore it for now with a comment or something?

Comment (CodeWithEmad) [reply setup.py]: yeah, we can do that if that's what you want.

File: tutorxqueue/plugin.py [hunk 7]
```diff
@@ -135,29 +140,43 @@ def show_submission(context):
...
-        grade,
-        correct,
-        message,
+        (
+            submission_id,
+            submission_key,
+            grade,
+            correct,
+            message,
+        ),
```
Comment (jfavellar90) [inline tutorxqueue/plugin.py:160]: @CodeWithEmad what is the purpose of grouping these 5 variables in the `print_result` method?

Comment (CodeWithEmad) [reply tutorxqueue/plugin.py:160]: I was trying to fix the typing issue where `*args: tuple[Any, ...]`. It may not be the optimal solution, but I couldn't find any other alternative.

**Comment by jfavellar90** [comment]
@CodeWithEmad thanks for your valuable contribution! There's just one comment to be addressed from my side. For the rest, it looks good to me :)

**Review by CodeWithEmad — COMMENTED** [review]
Oh! My comments were pending for 2 week! I hate it when Github does that!

**Comment by regisb** [comment]
Can we merge this today? I'd like to merge as many PRs as possible before Quince, on Monday.
